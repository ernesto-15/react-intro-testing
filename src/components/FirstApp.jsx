import React from 'react';
import PropTypes from 'prop-types';

const FirstApp = ({ hi, sub }) => {
  return (
    <>
      <h1>{hi}!!</h1>
      <p>{sub}</p>
    </>
  );
};

FirstApp.propTypes = {
  hi: PropTypes.string.isRequired,
};

FirstApp.defaultProps = {
  sub: 'I am a paragraph',
};

export default FirstApp;
