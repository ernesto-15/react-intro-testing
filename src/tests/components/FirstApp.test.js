// import { render } from '@testing-library/react';
import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import FirstApp from '../../components/FirstApp';

describe('tests in FirstApp', () => {
  // test('should show message "Hello Ernesto"', () => {
  //   const hi = 'Hello Ernesto';
  //   const { getByText } = render(<FirstApp hi={'Hello Ernesto'} />);
  //   expect(getByText(hi) + '!').toBeInTheDocument();
  // });

  test('should show FirstApp component', () => {
    const wrapper = shallow(<FirstApp hi="Hello Ernesto" />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should show sub sent from props', () => {
    const hi = 'Hello Ernesto';
    const sub = 'Testiiiiing!!!!!!!';
    const wrapper = shallow(<FirstApp hi={hi} sub={sub} />);
    const paragraphText = wrapper.find('p').text();
    expect(paragraphText).toBe(sub);
  });
});
