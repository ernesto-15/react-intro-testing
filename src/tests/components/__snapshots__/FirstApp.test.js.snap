// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests in FirstApp should show FirstApp component 1`] = `
<Fragment>
  <h1>
    Hello Ernesto
    !!
  </h1>
  <p>
    I am a paragraph
  </p>
</Fragment>
`;
